src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -426,6 +399,91 @@ impl ConcatSstableIterator {
         }
         Ok(())
     }
+
+    pub fn filter_block_metas(
+        block_metas: &Vec<BlockMeta>,
+        seek_key: Option<FullKey<&[u8]>>,
+        existing_table_ids: &HashSet<u32>,
+        key_range: &KeyRange,
```

Comment:
It seems that we assume `seek_key >= key_range.left` and we ignore `key_range.left` in this method. Would it be cleaner if we pass a correct key range for filtering without seek_key here?

Reply:
Good catch, refactor it